spec/ndx-patterned-ogen.extensions.yaml

```
@@ -1,8 +1,165 @@
 groups:
-- neurodata_type_def: TetrodeSeries
-  neurodata_type_inc: ElectricalSeries
-  doc: An extension of ElectricalSeries to include the tetrode ID for each time series.
-  attributes:
-  - name: trode_id
-    dtype: int32
-    doc: The tetrode ID.
+  - neurodata_type_def: OptogeneticStimulusPattern
+    neurodata_type_inc: LabMetaData
+    doc: Container to store the information about a generic stimulus pattern (spatial information).
+    attributes:
+      - name: description
+        dtype: text
+        doc: Description of the scanning or scanless method for shaping optogenetic light. Examples include diffraction limited points, 3D shot, disks, etc.
+        required: false
+      - name: sweep_size
+        dtype: numeric
+        doc:
+          Size of the scanning sweep pattern in micrometers. If a scalar is provided, the sweep pattern is
+          assumed to be a circle (for 2D patterns) or cylinder (for 3D patterns), with diameter 'sweep_size'.
+          If 'sweep_size' is a two or three dimensional array, the the sweep pattern is assumed to be a
+          rectangle or cuboid, with dimensions [width, height] or [width, height, depth].
+        required: false
+      - name: sweep_mask
+        dtype: numeric
+        doc:
+          Scanning sweep pattern designated using a mask of size [width, height] (2D stimulation) or [width,
+          height, depth] (3D stimulation), where for a given pixel a value of 1 indicates stimulation, and a
+          value of 0 indicates no stimulation.
+        required: false
+  - neurodata_type_def: SpiralScanning
+    neurodata_type_inc: LabMetaData
+    doc: Container to store the parameters defining a spiral scanning pattern.
+    attributes:
+      - name: description
+        dtype: text
+        doc: Description of the pattern.
+        required: false
+      - name: diameter
+        dtype: numeric
+        doc: Spiral diameter of each sweep (in micrometers).
+      - name: height
+        dtype: numeric
+        doc: Spiral height of each sweep (in micrometers).
+        required: false
+      - name: number_of_revolutions
+        dtype: numeric
+        doc: Number of turns within a spiral.
```

```suggestion
      - name: diameter
        dtype: numeric
        doc: Spiral diameter of each sweep (in micrometers).
        required: true
      - name: number_of_revolutions
        dtype: numeric
        doc: Number of turns within a spiral.
        required: true
      - name: description
        dtype: text
        doc: Description of the pattern.
        required: false
      - name: height
        dtype: numeric
        doc: Spiral height of each sweep (in micrometers).
        required: false
```

Two suggestions; move required fields to above the non-required ones; specify `required: true` even though it's the default, just to maximize readability without having to assume knowledge of how NWB schema treats the field

---

If you agree with this improvement, the rest of the spec here could be updated with the approach as well

---

Sure, thanks for taking the time to correct these. I will update the rest